product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org.wso2.carbon.ei.scenario.test/AddChildToXMLTest.java

```diff
@@ -0,0 +1,108 @@
+
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.ei.scenario.test;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.jaxen.JaxenException;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.http.SOAPClient;
+import org.wso2.carbon.esb.scenario.test.common.utils.FileUtils;
+import org.wso2.carbon.esb.scenario.test.common.utils.XMLUtils;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This test class contains tests related to Enrich Mediator being used to enrich payload with child elements.
+ */
+public class AddChildToXMLTest extends ScenarioTestBase {
+
+    private static final Log log = LogFactory.getLog(AddChildToXMLTest.class);
+    private String sourcesFilePath;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        sourcesFilePath = testResourcesDir + File.separator + "source_files";
+    }
+
+    @Test(description = "1.6.1.1 - [SOAP > POX] enrich a inline content as a child")
+    public void testAddingInlineContentAsChild() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_1_1_Proxy_SoapToPoxMsgEnrichWithChild");
+        String request = FileUtils.readFile(sourcesFilePath + File.separator + "request_1_6_1_1.xml");
+
+        Map<String, String> headers = new HashMap<>(1);
+        headers.put(ScenarioConstants.MESSAGE_ID, "1_6_1_1");
+
+        SOAPClient soapClient = new SOAPClient();
+        HttpResponse response = soapClient.sendSimpleSOAPMessage(url, request, "urn:mediate", headers);
+
+        String expectedResponse = FileUtils.readFile(sourcesFilePath + File.separator + "response_1_6_1_1.xml");
+
+        assertResponse(response, expectedResponse);
+    }
+
+
+    @Test(description = "1.6.1.2 - [SOAP > POX] enrich xpath element a child of the body")
+    public void testPassThroughProxyTemplate() throws IOException, XMLStreamException {
+
+        String url = getProxyServiceURLHttp("1_6_1_2_Proxy_SoapToPoxEnrichWithXpathAsChild");
+
+        String request = FileUtils.readFile(sourcesFilePath  + File.separator + "request_1_6_1_2.xml");
```

Comment:
Why are we reading from files? we can hard code in the test case since we cannot test this for multiple requests. That will improve readability and easily understand the test case.